### PR TITLE
Add DID rotation

### PR DESCRIPTION
**Pull request by rodolfomiranda**
Adding DID Rotation section

### CODE REVIEW

File: site/content/book/v2/didrotation.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  - `iat`: datetime in seconds
+- Signature: from the previous DID and key defined in the `kid`
+
+Once a DIDComm agent receives a message from an unknown DID, it must:
+1. check if there’s a `from_prior` header
+2. if exists, extract the DID from the `iss` and validate if it matches a known DID
+3. validate JWT signatures using the key defined in `kid` and extracted from previous DID defined in `iss`
+4. if all validations are successful, the new DID should be stored for future communications with the previously known agent
+
+Example code showin a DID Rotation after an Out of Band message can be found in the section [Starting, using and ending a DIDComm connection](startConnection) from this Guidebook.
```
Comment (2byrds) [inline site/content/book/v2/didrotation.md:27]: (typo fix) Example code showing

File: site/content/book/v2/didrotation.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+Since DIDComm is asynchronous in nature, messages can arrive in different order. Sender and receiver should be care of:
```
Comment (2byrds) [inline site/content/book/v2/didrotation.md:29]: (typo fix) Sender and receiver should take care to:

File: site/content/book/v2/didrotation.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+- Sender: avoid starting a DID rotation in the middle of multiple message conversations. If DID rotation arrives out of order, some messages may be discarded by the recipient.
+- Receiver: messages received from the old DID but after the message that performs the DID rotation must be discarded to reduce the risk of potential compromised keys.
```
Comment (2byrds) [inline site/content/book/v2/didrotation.md:31]: (typo fix) risk of potentially compromised keys.

**Review by 2byrds — CHANGES_REQUESTED** [review]
Great stuff!  Learned a ton.  Typo fixes requested.

File: site/content/book/v2/didrotation.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+## DID Rotation
+
+[DIDComm Messaging](https://identity.foundation/didcomm-messaging/spec/) relies on the [DIDs](https://www.w3.org/TR/did-core/), and associated [DID Documents](https://www.w3.org/TR/did-core/#dfn-did-documents), of the two peers involved in the communication. Those parties are responsible to secure their keys and perform key rotations if they are compromised, and to make other changes in the DID Documents if needed such as service endpoints. DIDComm has no involvement in that procedure that is particular to each DID method; nevertheless and provided that the DID does not change, the communication can keep flowing normally.
```
Comment (TelegramSam) [inline site/content/book/v2/didrotation.md:3]: This is a minor nit, but key rotations can and should happen even if they are not compromised. Adding just a bit to acknowledge that will prevent it from detracting from the larger work you've contributed.

File: site/content/book/v2/didrotation.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+However there are cases when the DID (not the DID Document) need to be rotated by one of the parties, such as in the following two cases:
+1. At the beginning of a connection: its common that the initial message is in the form of an Out of Band message that it is unencrypted per definition, and may be observed by another party or transmitted in an unsecure channel such as a QR code or a URL posted in an email or webpage. The DID used in the OOB should be considered as a temporal DID just to start the conversation, but it is highly recommended that it is rotated afterwards to improve privacy.
+2. When keys inside a DID Document are compromised and the DID method does not allow updating the DID Document with new rotated keys, as is the case for [did:key](https://w3c-ccg.github.io/did-method-key/).
+
```
Comment (TelegramSam) [inline site/content/book/v2/didrotation.md:8]: Another reason is needing or wanting to move from one DID method to another.

**Comment by TelegramSam** [comment]
I love this! thanks for your work. It might be worth a mention that the party doing the rotation should continue to send the rotation header until they receive a message from the other party using the new DID. This allows for continuity even if one party is offline or not processing messages for a bit.

**Comment by rodolfomiranda** [comment]
In last commit I corrected the typos and added @TelegramSam suggestions.

**Review by dhh1128 — APPROVED** [review]

**Pull request merged**